tests/integration/base.py

```diff
@@ -278,6 +279,15 @@ def start_dnsmasq(self, ipv6_mode, iface):
         else:
             self.poll_text(dnsmasq_log, 'DHCP, IP range')
 
+    def iface_json(self, iface: str) -> dict:
+        '''Return iproute2's (detailed) JSON representation'''
+        out = subprocess.check_output(['ip', '-j', '-d', 'a', 'show', 'dev', iface],
```

Comment:
question: should we handle a possible call against an interface that doesn't exist?

Reply:
I don't think that's necessarily needed. This is only part of the integration tests, so all we could do is: bail out.
If the interface doesn't exist, this will crash with an error message, too, as-is.

IMHO, no need to over-engineer this, but rather keep it small and simple in the test framework. All the input is controlled, as it originates from our own test scripts, not random user input.